source/DataFile.h

```diff
@@ -16,6 +16,7 @@ this program. If not, see <https://www.gnu.org/licenses/>.
 #ifndef DATA_FILE_H_
 #define DATA_FILE_H_
 
+#include "ConditionsStore.h"
```

Comment:
Previously, the `Load()` functions had to have a ConditionsStore so they could pass it to DataNode.AsCondition. Putting it inside the DataFile shrank the code considerably. The alternative is to only allow DataNodes to connect to the universe conditions, but that would be problematic if, for example, the ConditionsStore should be the global variables or the player variables.

Reply:
> Conditions as used in missions are currently also loaded from datafiles, and somehow they don't need this functionality.

No, they don't. The names are loaded from the datafile, and the code contacts a ConditionsStore later on to get the condition value. If that was done in deeply nested loops inside the engine, the code would be very slow due to the repeated string hashes and map lookups. Even now, dictionary lookups take as much as 30% of the runtime in busy systems. Let's not make that worse.

The Condition class and UpdateConditions optimizes this, limiting hashes and map lookups to the bare minimum necessary.

It would be possible to change the code to make Condition objects without a ConditionsStore, and then connect them later on. To do that, something would have to tell every Condition to get its store, OR the Condition would have to be limited to getting one specific store, like the universe's conditions.